Project2.ipynb

```diff
@@ -272,38 +272,118 @@
     "import tensorflow_datasets as tfds\n",
     "import tensorflow_probability as tfp\n",
     "from tensorflow.keras import layers\n",
+    "from kerastuner.tuners import RandomSearch, BayesianOptimization\n",
+    "from kerastuner.engine.hyperparameters import HyperParameters\n",
+    "import time\n",
+    "from numpy.random import seed\n",
+    "seed(1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Hyper parameter tuning\n",
     "\n",
-    "## A feedforward neural network with one hidden layer is made. A ReLU activation is used in the hidden layer.\n",
-    "## An output layer with a single output node is used\n",
+    "Hyperparameter tuning of a NN model using either Random Search or Bayesian optimization:\n",
     "\n",
-    "# Number of neurons in hidden layer:\n",
-    "n_neurons = 10\n",
+    "The hyperparameters are:\n",
     "\n",
-    "# L2 regularization \"lambda\"\n",
-    "lmbda = 0\n",
+    "1. Input layer size\n",
+    "2. Number of layers\n",
+    "3. Size of each layer\n",
+    "4. Regularization term (same for each layer)\n",
+    "5. Learning rate\n",
+    "6. Learning rate decay\n",
+    "\n",
+    "The maximum values, minimum values and step size are chosen below in the method ```build_model```\n",
+    "\n",
+    "\n",
+    "# Next step might take a long time to run!!!\n",
```

Comment:
How long does it take on your machine? (secodns/minutes/hours or days?)

Reply:
Did I forget to write it? It took 2 hours and 22 minutes for 1000 trials. I did change the number of trials to 10 before I added it to git.